internal/jobs/linkedin.go

```diff
@@ -0,0 +1,289 @@
+	if err != nil {
+		// Check for specific error types
+		if strings.Contains(err.Error(), "unauthorized") || strings.Contains(err.Error(), "401") {
+			ls.statsCollector.Add(j.WorkerID, stats.LinkedInAuthErrors, 1)
+		} else if strings.Contains(err.Error(), "rate limit") || strings.Contains(err.Error(), "429") {
+			ls.statsCollector.Add(j.WorkerID, stats.LinkedInRateErrors, 1)
+		} else {
+			ls.statsCollector.Add(j.WorkerID, stats.LinkedInErrors, 1)
+		}
+		return types.JobResult{Error: fmt.Sprintf("failed to search profiles: %v", err)}, err
+	}
+
+	// Convert to our result type
+	var results []teetypes.LinkedInProfileResult
+	for _, profile := range profiles {
+		result := teetypes.LinkedInProfileResult{
+			PublicIdentifier: profile.PublicIdentifier,
+			URN:              profile.URN,
+			FullName:         profile.FullName,
+			Headline:         profile.Headline,
+			Location:         profile.Location,
+			ProfileURL:       profile.ProfileURL,
+			// Degree field will be empty for now since BadgeText is not available
+			Degree: "",
+		}
+		results = append(results, result)
+	}
+
+	ls.statsCollector.Add(j.WorkerID, stats.LinkedInProfiles, uint(len(results)))
+
+	data, err := json.Marshal(results)
+	if err != nil {
+		return types.JobResult{Error: "failed to marshal results"}, err
+	}
+
+	return types.JobResult{Data: data}, nil
+}
+
+func (ls *LinkedInScraper) getProfile(j types.Job, client *linkedinscraper.Client, args *args.LinkedInArguments) (types.JobResult, error) {
+	// Validate public identifier is not empty
+	if args.PublicIdentifier == "" {
+		ls.statsCollector.Add(j.WorkerID, stats.LinkedInErrors, 1)
+		return types.JobResult{Error: "public_identifier is required"}, fmt.Errorf("public_identifier is required")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), j.Timeout)
+	defer cancel()
+
+	profile, err := client.GetProfile(ctx, args.PublicIdentifier)
+	if err != nil {
+		// Check for specific error types
+		if strings.Contains(err.Error(), "unauthorized") || strings.Contains(err.Error(), "401") {
+			ls.statsCollector.Add(j.WorkerID, stats.LinkedInAuthErrors, 1)
+		} else if strings.Contains(err.Error(), "rate limit") || strings.Contains(err.Error(), "429") {
+			ls.statsCollector.Add(j.WorkerID, stats.LinkedInRateErrors, 1)
+		} else if strings.Contains(err.Error(), "not found") || strings.Contains(err.Error(), "404") {
+			ls.statsCollector.Add(j.WorkerID, stats.LinkedInErrors, 1)
+		} else {
+			ls.statsCollector.Add(j.WorkerID, stats.LinkedInErrors, 1)
+		}
+		return types.JobResult{Error: fmt.Sprintf("failed to get profile: %v", err)}, err
+	}
+
+	// Use the new LinkedInFullProfileResult structure for rich profile data
+	// Convert LinkedIn scraper types to tee-types
+	var experiences []teetypes.Experience
+	for _, exp := range profile.Experience {
+		// Convert DateRange to string format for tee-types
+		var startDate, endDate string
+		if exp.DateRange != nil {
```

Comment:
Nit: Since we're using this multiple times, extract it into a function to ensure that we always use the same format.